tests/saes/test_transcoder.py

```diff
@@ -0,0 +1,352 @@
+
+        # Check that features are sparse (ReLU should zero out some)
+        assert (features == 0).any()
+
+    def test_transcoder_encode_decode(self):
+        cfg = build_transcoder_cfg()
+        transcoder = Transcoder(cfg)
+
+        batch_size = 5
+        x = torch.randn(batch_size, cfg.d_in)
+
+        # Test encode
+        features = transcoder.encode(x)
+        assert features.shape == (batch_size, cfg.d_sae)
+
+        # Test decode
+        output = transcoder.decode(features)
+        assert output.shape == (batch_size, cfg.d_out)
+
+    def test_transcoder_process_sae_in(self):
+        """Test that process_sae_in doesn't apply b_dec (wrong dimension)."""
+        cfg = build_transcoder_cfg()
+        transcoder = Transcoder(cfg)
+
+        # Set b_dec to non-zero values
+        transcoder.b_dec.data = torch.ones_like(transcoder.b_dec) * 5.0
+
+        x = torch.randn(10, cfg.d_in)
+        processed = transcoder.process_sae_in(x)
+
+        # Should be same as input (just dtype conversion and hooks)
+        assert_close(processed, x.to(transcoder.dtype))
+
+    def test_transcoder_get_output_hook_name(self):
+        # Without head index
+        cfg = build_transcoder_cfg(hook_name_out="blocks.1.hook_mlp_out")
+        transcoder = Transcoder(cfg)
+        assert transcoder.get_output_hook_name() == "blocks.1.hook_mlp_out"
+
+        # With head index
+        cfg_with_head = build_transcoder_cfg(
+            hook_name_out="blocks.1.attn.hook_z", hook_head_index_out=3
+        )
+        transcoder_with_head = Transcoder(cfg_with_head)
+        assert transcoder_with_head.get_output_hook_name() == "blocks.1.attn.hook_z.h3"
+
+    def test_transcoder_from_dict(self):
+        config_dict = {
+            "d_in": 256,
+            "d_sae": 512,
+            "d_out": 384,
+            "dtype": "float32",
+            "device": "cpu",
+            "hook_name_out": "test_out",
+            "architecture": "transcoder",
+        }
+
+        transcoder = Transcoder.from_dict(config_dict)
+        assert isinstance(transcoder, Transcoder)
+        assert transcoder.cfg.d_in == 256
+        assert transcoder.cfg.d_out == 384
+
+    def test_transcoder_to_device(self):
+        cfg = build_transcoder_cfg(device="cpu")
+        transcoder = Transcoder(cfg)
+
+        # Move to meta device
+        transcoder_moved = transcoder.to("meta")
+        assert transcoder_moved.device == torch.device("meta")
+        assert transcoder_moved.W_enc.device == torch.device("meta")
+        assert transcoder_moved.W_dec.device == torch.device("meta")
+        assert transcoder_moved.b_enc.device == torch.device("meta")
+        assert transcoder_moved.b_dec.device == torch.device("meta")
+
+    def test_transcoder_fold_w_dec_norm(self):
+        cfg = build_transcoder_cfg()
+        transcoder = Transcoder(cfg)
+
+        # Set weights to non-unit norm
+        transcoder.W_dec.data = torch.randn_like(transcoder.W_dec) * 2.0
```

Comment:
We should ensure every parameter is set randomly, if anything is zeros this test might pass while silently breaking the underlying transcoder. See `test_sae_fold_w_dec_norm_all_architectures` for an example of how to test this while initializing all parameters randomly.

Alternatively, maybe it's better to add `transcoder`, `jumprelu_transcoder`, and `skip_transcoder` to the `ALL_ARCHITECTURES` list in tests so these automatically get tested with the rest of the SAE classes. Or maybe we could make a new `ALL_ARCHITECTURES_INCLUDING_TRANSCODERS` list or something for tests that apply to everything like this.